### PR TITLE
Deprecate `add-snippets` input.

### PR DESCRIPTION
**Pull request by rvermeulen**
### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ Note that the only difference between `v2` and `v3` of the CodeQL Action is the
 
 - _Deprecation:_ Swift analysis on Ubuntu runner images is no longer supported. Please migrate to a macOS runner if this affects you. [#2403](https://github.com/github/codeql-action/pull/2403)
 - Bump the minimum CodeQL bundle version to 2.13.5. [#2408](https://github.com/github/codeql-action/pull/2408)
+- _Deprecation:_ The `add-snippets` input on the `analyze` Action is deprecated and will be removed in the first release in August 2025.
```
Comment (aeisenberg) [inline CHANGELOG.md:11]:
```suggestion
- _Deprecation:_ The `add-snippets` input on the `analyze` Action is deprecated and will be removed in the first release in August 2025. [#2436](https://github.com/github/codeql-action/pull/2436)
```

**Review by aeisenberg — APPROVED** [review]
LGTM after adding the link to the PR in the changelog. Also, can you create a deprecation issue in our private repository to track the deprecation?

**Review by aeisenberg — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -24,6 +24,7 @@ No user facing changes.
 
 - _Deprecation:_ Swift analysis on Ubuntu runner images is no longer supported. Please migrate to a macOS runner if this affects you. [#2403](https://github.com/github/codeql-action/pull/2403)
 - Bump the minimum CodeQL bundle version to 2.13.5. [#2408](https://github.com/github/codeql-action/pull/2408)
+- _Deprecation:_ The `add-snippets` input on the `analyze` Action is deprecated and will be removed in the first release in August 2025. [#2436](https://github.com/github/codeql-action/pull/2436)
```
Comment (henrymercer) [inline CHANGELOG.md:27]: Did you intentionally add this to 3.26.0?

Comment (rvermeulen) [reply CHANGELOG.md:27]: I added it to unreleased, must have gone wrong when I accepted a review suggestion 🤔. Let me move it back to unreleased.

**Review by henrymercer — APPROVED** [review]

**Pull request merged**